src/App.js

```diff
@@ -1,14 +1,142 @@
+    const OMBbAPIKey = "aba7af8e";
+    const query = 'http://www.omdbapi.com/?s=' + this.state.searchTerm + '&apikey=' + OMBbAPIKey + '&page=' + this.state.page;
+
+    // Back to top button
+    if (body.scrollTop > 300 || html.scrollTop > 300) {
+      backToTop.classList.add('visible');
+    } else {
+      backToTop.classList.remove('visible');
+    }
+
+    // Infinite scroll
+    if (windowBottom >= docHeight) {
+      fetch(query)
+        .then(response => {
+          loading.textContent = "Loading...";
+          loading.classList.add('show');
+          return response.json();
+        }
+        )
+        .then(movies => {
+          if (movies.Search) {
+            const self = this;
+            setTimeout(function () {
+              self.setState({
+                search: [...self.state.search, ...movies.Search],
+                page: self.state.page += 1
+              });
+              loading.classList.remove('show');
+            }, 500)
+          } else {
+            loading.textContent = "No more results";
+            loading.classList.add('show');
+          }
+        })
+        .catch(error => {
+          console.log(error);
+          loading.textContent = "There's been a problem fetching the results. Try again later..";
+          loading.classList.add('show');
+        });
+
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", this.handleScroll);
+  }
+
+  receiverSearch(movies, searchTerm) {
+    this.setState({
+      search: movies,
+      searchTerm: searchTerm,
+      watchListSection: false
+    })
+  }
+
+  receiverDisplayWatchlist() {
+    this.setState({
+      watchListSection: true
+    })
+  }
+
+  receiverWatchList(articleTitle, action) {
+    if (action === "addToWatchList") {
+      this.state.search.forEach(articleObj => {
+        return Object.keys(articleObj).find(articleKey => {
+          if (articleObj[articleKey] === articleTitle) {
+            this.setState({
+              watchList: [...this.state.watchList, articleObj]
+            })
+
+            localStorage.getItem("watchList")
+              ? localStorage.setItem("watchList", JSON.stringify([articleObj, ...JSON.parse(localStorage.getItem("watchList"))]))
+              : localStorage.setItem("watchList", JSON.stringify([...this.state.watchList, articleObj]));
+          }
+        });
+      });
+    } else {
+      const index = JSON.parse(this.state.watchList).forEach((item, index) => {
+        if (item.Title === articleTitle) {
+          // console.log("Aquis", this.state.watchList);
+          this.setState({
+            watchList: JSON.parse(this.state.watchList).splice(index, 1)
+          })
+          localStorage.setItem("watchList", this.state.watchList);
+          // console.log("Aquis", localStorage.getItem("watchList"));
+        }
+      });
+    }
   }
 
-  render(){
+  render() {
+    const watchList = this.state.watchListSection;
+    let MoviesList;
```

Comment:
avoid using capitals for things which are not classes or component names

Reply:
I understand that in this case it will contain a component instance, but it will be an instance of the class rather than the class itself. For example

```js
const movie = new Movie();
```